server: TestServerStartClock is flaky

```
> make stress PKG='./pkg/server' TESTS=TestServerStartClock

--- FAIL: TestServerStartClock (0.19s)
    server_test.go:161: time: server 1547494457.648957000,1 vs actual 1547494457648957000

ERROR: exit status 1

88 runs completed, 1 failures, over 4s
context canceled
FAIL
```

--- a/pkg/server/server_test.go
+++ b/pkg/server/server_test.go
@@ -153,10 +153,8 @@ func TestServerStartClock(t *testing.T) {
 	}
 
 	now := s.Clock().Now()
-	// We rely on s.Clock() having been initialized from hlc.UnixNano(), which is a
-	// bit fragile.
-	physicalNow := hlc.UnixNano()
-	serverClockWasPushed := (now.Logical > 0) || (now.WallTime > physicalNow)
+	physicalNow := s.Clock().PhysicalNow()
+	serverClockWasPushed := now.WallTime > physicalNow
 	if serverClockWasPushed {
 		t.Fatalf("time: server %s vs actual %d", now, physicalNow)
 	}

